tests/test_rst_compiler.py

```diff
@@ -1,238 +1,227 @@
-always unquoted.
+    * assertHTMLContains(html, element, attributes=None, text=None)
 
 """
 
-import os
-
 import io
-try:
-    from io import StringIO
-except ImportError:
-    from StringIO import StringIO  # NOQA
-import tempfile
+import os
+from io import StringIO
 
 import docutils
-from lxml import html
-import unittest
+from lxml import html as lxml_html
 
 import nikola.plugins.compile.rest
-from nikola.plugins.compile.rest import vimeo
 import nikola.plugins.compile.rest.listing
-from nikola.plugins.compile.rest.doc import Plugin as DocPlugin
-from nikola.utils import _reload
-from .base import BaseTestCase, FakeSite, FakePost
-
-
-class ReSTExtensionTestCase(BaseTestCase):
-    """ Base class for testing ReST extensions """
-
-    sample = 'foo'
-    deps = None
-
-    def setUp(self):
-        self.compiler = nikola.plugins.compile.rest.CompileRest()
-        self.compiler.set_site(FakeSite())
-        return super(ReSTExtensionTestCase, self).setUp()
-
-    def basic_test(self):
-        """ Parse cls.sample into a HTML document tree """
-        self.setHtmlFromRst(self.sample)
-
-    def setHtmlFromRst(self, rst):
-        """ Create html output from rst string """
-        tmpdir = tempfile.mkdtemp()
-        inf = os.path.join(tmpdir, 'inf')
-        outf = os.path.join(tmpdir, 'outf')
-        with io.open(inf, 'w+', encoding='utf8') as f:
-            f.write(rst)
-        p = FakePost('', '')
-        p._depfile[outf] = []
-        self.compiler.site.post_per_input_file[inf] = p
-        self.html = self.compiler.compile(inf, outf)
-        with io.open(outf, 'r', encoding='utf8') as f:
-            self.html = f.read()
-        os.unlink(inf)
-        os.unlink(outf)
-        depfile = [p for p in p._depfile[outf] if p != outf]
-        depfile = '\n'.join(depfile)
-        if depfile:
-            self.assertEqual(self.deps.strip(), depfile)
-        os.rmdir(tmpdir)
-        self.html_doc = html.parse(StringIO(self.html))
-
-    def assertHTMLContains(self, element, attributes=None, text=None):
-        """ Test if HTML document includes an element with the given
-        attributes and text content
-
-        """
-        try:
-            tag = next(self.html_doc.iter(element))
-        except StopIteration:
-            raise Exception("<{0}> not in {1}".format(element, self.html))
-        else:
-            if attributes:
-                arg_attrs = set(attributes.items())
-                tag_attrs = set(tag.items())
-                self.assertTrue(arg_attrs.issubset(tag_attrs))
-            if text:
-                self.assertIn(text, tag.text)
-
-
-class ReSTExtensionTestCaseTestCase(ReSTExtensionTestCase):
-    """ Simple test for our base class :) """
+from nikola.plugins.compile.rest import vimeo
+from nikola.utils import _reload, LocaleBorg
+
+from .base import FakeSite, FakePost
+
+import pytest
+
+
+def test_ReST_extension(tempdir):
+    "Simple test for our base class :)"
```

Comment:
The docstring is probably outdated.  Also, we prefer `"""triple quotes"""` for docstrings.

Reply:
There will be a few other places where single quotes are in place. I'll check for these!

Reply:
You could try to use the `black` automatic formatter on tests only. It’s a great tool overall; converting all of Nikola to `black` would make a huge mess, but since you’re editing practically all files under `tests/`, it will work out here nicely. Feel free to pick a length limit greater than the default (88).